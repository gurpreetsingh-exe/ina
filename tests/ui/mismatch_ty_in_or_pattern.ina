// COMPILE_FAIL

type Foo =
    | A(i32)
    | B(i64)

fn main() {
    match Foo::A(20) {
        Foo::A(x) | Foo::B(x) => x,
    };
}
