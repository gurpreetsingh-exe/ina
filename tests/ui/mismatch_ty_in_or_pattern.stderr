error: mismatch types
   ╭─[tests/ui/mismatch_ty_in_or_pattern.ina:9:21]
   │
 9 │         Foo::A(x) | Foo::B(x) => x,
   │                     ───┬──
   │                        ╰──── expected `i64`, found `i32`
   ∙
───╯
